internal/encoding/gotypes/generate.go

```diff
@@ -100,7 +109,7 @@ func Generate(ctx *cue.Context, insts ...*build.Instance) error {
 			buf = fmt.Appendf(buf, format, args...)
 		}
 		printf("// Code generated by \"cue exp gengotypes\"; DO NOT EDIT.\n\n")
-		goPkgName := goPkgNameForInstance(inst, instVal)
+		goPkgName := goPkgNameForInstance(inst, instVal, g.importAliases)
```

Comment:
this is rather crucial - you must NOT use `importAliases` here, so you should use a `nil` map instead. this tracks which Go packages (by name) we generate per directory, as we can only generate one package per directory. This is entirely unrelated to what aliases a package is imported under.